finagle-core/src/main/scala/com/twitter/finagle/loadbalancer/TrafficDistributor.scala

```diff
@@ -272,12 +277,9 @@ private class TrafficDistributor[Req, Rep](
   // ServiceFactories that can service requests.
   private[this] val underlying: Event[ServiceFactory[Req, Rep]] =
     weightClasses.foldLeft(init) {
-      case (_, Activity.Ok(wcs)) if wcs.isEmpty =>
-        // Defer the handling of an empty destination set to `newBalancer`
-        val emptyBal = newBalancer(Activity(Var(Activity.Ok(Set.empty[EndpointFactory[Req, Rep]]))))
-        updateGauges(wcs)
-        pending.updateIfEmpty(Return(emptyBal))
-        emptyBal
+      case (_, Activity.Ok(List(WeightClass(lb, _, _,sz)))) if sz == 0 =>
```

Comment:
do we still need this case? I think the `Distributor` should just do the right thing.